packages/graph/hash_graph/lib/graph/src/api/rest/entity.rs

```diff
@@ -96,6 +97,10 @@ struct CreateEntityRequest {
     owned_by_id: OwnedById,
     entity_id: Option<EntityId>,
     actor_id: CreatedById,
+    // TODO: this could break invariants if we don't move to fractional indexing
+    //  https://app.asana.com/0/1201095311341924/1202085856561975/f
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    link_metadata: Option<LinkEntityMetadata>,
```

Comment:
If we are worried about the ordering, we could put a unique constraint on the columns. That way we move the integrity check to the database at least

Reply:
I'm not entirely sure how that would work with the histories table. I also am not sure we could do it _only_ in the latest table.. should an archived link still be in the order? What happens if you try and unarchive a link and it has a clash in index

Reply:
Continuing discussion offline, potentially for follow-up. We're thinking we _maybe_ don't need to guarantee stable sorts in the first place.